### ISSUE TITLE
After drawing more sophisticated shape, there's an exception in console:

### ISSUE THREAD

**Issue by adammoszczynski** [Wildhoney/Leaflet.FreeDraw#128]
I tried it on your demo on jsfiddle.

Exception is as follows:
leaflet-freedraw.web.js:23052 Uncaught RangeError: Maximum call stack size exceeded
    at Generator.invoke [as _invoke] (leaflet-freedraw.web.js:22774)
    at Generator.prototype.(fiddle.jshell.net/_display/anonymous function) [as return] (https://rawgit.com/Wildhoney/Leaflet.FreeDraw/master/dist/leaflet-freedraw.web.js:22596:21)
    at tryCatch (leaflet-freedraw.web.js:22544)
    at maybeInvokeDelegate (leaflet-freedraw.web.js:22844)
    at Generator.invoke [as _invoke] (leaflet-freedraw.web.js:22756)
    at Generator.prototype.(fiddle.jshell.net/_display/anonymous function) [as return] (https://rawgit.com/Wildhoney/Leaflet.FreeDraw/master/dist/leaflet-freedraw.web.js:22596:21)
    at tryCatch (leaflet-freedraw.web.js:22544)
    at maybeInvokeDelegate (leaflet-freedraw.web.js:22844)
    at Generator.invoke [as _invoke] (leaflet-freedraw.web.js:22756)
    at Generator.prototype.(fiddle.jshell.net/_display/anonymous function) [as return] (https://rawgit.com/Wildhoney/Leaflet.FreeDraw/master/dist/leaflet-freedraw.web.js:22596:21)

**Comment by Wildhoney** [comment]
How do I replicate it? Browser and version? What is a *complex* shape?

**Comment by adammoszczynski** [comment]
Chrome version:72.0.3626.121

Screenshot just before releasing mouse button:

![shape](https://user-images.githubusercontent.com/48747902/54669333-63982580-4af1-11e9-892f-2f5638203416.jpg)

I think it is important to zoom out to the zoom as on screenshot.

**Comment by Wildhoney** [comment]
I tracked it down to the recursive nature of the generators, which aren't tail call optimised.

I merged [the pending PR](https://github.com/Wildhoney/Leaflet.FreeDraw/pull/115) which eliminates the need for the generators, which has improved it significantly, so I've just published a new release.

However it does get more sluggish at around 10,000 iterations of the `mouseMove` function, so I may need to look further into actually optimising the recursive function to take better advantage of TCO.
